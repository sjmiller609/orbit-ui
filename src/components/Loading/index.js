import React from 'react'

const Loading = ({ prop }) => {
  return (
    <div>Component</div>
  )
}

export default Loading
