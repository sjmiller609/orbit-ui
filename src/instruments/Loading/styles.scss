@import '../../styles/mixins.scss';

.loading {
	position: fixed;
	z-index: 99999;
	top: 0;
	right: 0%;
	height: .4rem;
	width: 100%;

	div {
		animation: loading 2s infinite ease-in-out;
		width: 100%;
		height: 100%;
		transform: translateX(-100vw);
		border-radius: 100px;

		background-image: $green; /* Old browsers */
	  background-image: linear-gradient(to right, transparent, $blue, $green); /* Standard syntax; must be last */
	}
}


@keyframes loading {
	0% {
		transform: translateX(-100vw);
		opacity: 1;
	}
	80% {
		transform: translateX(100vw);
	}
	81%, 100% {
		opacity: 0;
	}
}
