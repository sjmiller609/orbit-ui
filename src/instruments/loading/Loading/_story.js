import React from 'react'
import { storiesOf } from '@storybook/react'
import { Loading } from 'instruments'

storiesOf('Instruments|Loading', module).add('Default', () => <Loading />)
