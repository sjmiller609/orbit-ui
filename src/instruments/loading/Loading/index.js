'use strict'
import React from 'react'
import s from './styles.scss'
import { Helmet } from 'react-helmet'

const Loading = () => {
  return (
    <div className={s.loading}>
      <Helmet>
        <title>Loading...</title>
      </Helmet>
      <div />
    </div>
  )
}

export default Loading
