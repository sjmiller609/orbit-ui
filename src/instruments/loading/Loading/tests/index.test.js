import React from 'react'
import { shallow } from 'enzyme'
import Loading from '../index'

describe('Loading', () => {
  it('should be defined', () => {
    expect(Loading).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(<Loading />)
    expect(wrapper).toMatchSnapshot()
  })
})
