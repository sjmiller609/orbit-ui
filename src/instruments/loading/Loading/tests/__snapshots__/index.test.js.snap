// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  className="loading"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Loading...
    </title>
  </HelmetWrapper>
  <div />
</div>
`;
